src/commcare_cloud/ansible/roles/bootstrap-users/tasks/main.yml

```diff
@@ -64,10 +64,51 @@
   tags:
     - new_ansible_sudo_pass
 
+- name: lookup ssh pubkeys from authorized_keys_dir and create ssh_pubkeys_list
+  set_fact:
+    ssh_pubkeys_list: "{{ lookup('file', authorized_keys_dir ~ item ~ '.pub') }}"
+  with_items:
+    "{{ dev_users.present }}"
+  register: ssh_pubkeys_results_list
+  tags:
+    - pubkey
+
+- name: iterate over ssh_pubkeys_list and join into a string
+  set_fact:
+    ssh_pubkeys_string: "{{ ssh_pubkeys_results_list.results | map(attribute='ansible_facts.ssh_pubkeys_list') | list | join('\n') }}"
+  tags:
+    - pubkey
+
+- name: capturing username
+  pause:
+    prompt: "Please enter your Dimagi username using which you SSH into servers."
+  register: username_input
+  tags:
+    - capture-userinfo
+    - pubkey
+- debug:
+    msg: "Entered username: {{ username_input.user_input }}"
+  tags:
+    - capture-userinfo
+    - pubkey
```

Comment:
@dannyroberts I think this is related to https://github.com/dimagi/commcare-cloud/pull/4261#discussion_r605027191, is that right? To me this seems to violate the typical run behavior of ansible in a surprising and non-intuitive way. Except at the beginning when prompting for sudo password, for example, ansible does not usually stop in the middle of a run to prompt for input (and I'd be annoyed if it did). I think if we really need this information we should prompt for it in commcare-cloud before running ansible and/or provide a way for developers to put this information in a file similar to `./src/commcare_cloud/fab/config.py` to avoid the prompt.

I'm not convinced that the risk is even worth it though. We do a "check mode" run first by default to prevent mistakes like that. I don't think it's worth adding more inconvenience, not to mention code to maintain, to avoid a situation that is rarely a concern. If someone did accidentally remove their own key it would be easy enough to get another developer to help them add it again.

Reply:
Yeah it's related to that comment. @millerdev I agree with what you said about the awkwardness of it. In terms of risk, the outcome we want to avoid isn't so much someone removing just themselves (though that would be mildly annoying), it's more someone removing _everyone_ including themselves, or a large chunk of people including themselves, accidentally. We could still use SSM (on AWS) to get access, but with our current tools it would be a very manual/awkward process. It would require human error for sure, but I still think it's foreseeable enough and has bad enough an outcome that it's worth guarding against. I'm not married to this particular _way_ of guarding against it though.

Since we're already skipping the prompt for `deploy-stack --first-time`, and the only other place this likely to run is `update-users`, I agree with the feedback that we should prompt for this from the python code in `update-users` at the beginning rather than here in the middle of the ansible script.

Reply:
Thanks, @dannyroberts @millerdev for your suggestions. I will work on taking input from python and passing it to ansible.